### PR TITLE
new flag use_noise_suppression in XTandemAdapter

### LINKED ISSUE #1392: 'use noise suppression' missing from XTandemAdapter
The 'use noise suppression' flag [1] is missing from XTandemAdapter. It should be added to the adapter and switched off by default. Currently it's switched on.

< note type="input" label="spectrum, use noise suppression" >yes< /note >

[1] http://www.thegpm.org/TANDEM/api/

### PR DESCRIPTION
**Pull request by lars20070**
The use_noise_suppression flag is added to XTandemAdapter. Resolves #1392.

### CODE REVIEW

File: src/topp/XTandemAdapter.cpp [hunk 1]
```diff
@@ -349,6 +350,7 @@ class TOPPXTandemAdapter :
     infile.setCleavageSite(getStringOption_("cleavage_site"));
     infile.setNumberOfMissedCleavages(getIntOption_("missed_cleavages"));
     infile.setRefine(getFlag_("refinement"));
+    infile.setRefine(getFlag_("use_noise_suppression"));
```
Comment (timosachsenberg) [inline src/topp/XTandemAdapter.cpp]: setNoiseSuppression!

File: src/openms/source/FORMAT/XTandemInfile.cpp [hunk 2]
```diff
@@ -685,11 +692,21 @@ namespace OpenMS
     return refine_;
   }
 
+  bool XTandemInfile::isNoiseSuppression() const
```
Comment (cbielow) [inline src/openms/source/FORMAT/XTandemInfile.cpp]: the function name is weird.
I can see that it nicely fits to isRefining(), but useNoiseSuppression
or getNoiseSuppression would be preferred.
Minor note:
In fact, naming all functions getXY() and setXY() make it easier for developers who use and IDE with auto-completion.

**Comment by cbielow** [comment]
Also, from what I understand from http://www.thegpm.org/TANDEM/api/suns.html, if noise suppression == false, then other parameters set in the config XML of XTandem will be ignored (e.g.
http://www.thegpm.org/TANDEM/api/smpmh.html), we should mention this in the description of the noise suppression parameter.

**Comment by lars20070** [comment]
@cbielow The parameters that are ignored for 'false' are not accessible in the XTandemAdapter interface. So I would not mention them in the description.

When updating to PILEDRIVER #1379, we should expose all (!) parameters and allow for a tandem param file as optional (advanced) input.

**Comment by timosachsenberg** [comment]
+1

**Pull request merged**